test/unit/carousel.ts

```diff
@@ -0,0 +1,403 @@
+        carousel.moveNext = spy();
+
+        carousel.onTouchEnd(event as any);
+
+        expect(Carousel.isSwipeToNext).to.be.calledWithExactly({ startX: 0, startY: 5 }, 30, 15);
+        expect(event.target.removeEventListener).to.be.calledWithExactly(
+          'touchstart',
+          carousel.onTouchStart
+        );
+        expect(event.target.removeEventListener).to.be.calledWithExactly(
+          'touchend',
+          carousel.onTouchEnd
+        );
+        s.restore();
+      });
+
+      it('should not swipe slides if swipe distance is not long enough', () => {
+        const event = {
+          changedTouches: [
+            {
+              pageX: 10,
+              pageY: 15
+            }
+          ],
+          target: {
+            addEventListener: spy(),
+            removeEventListener: spy()
+          }
+        };
+        Carousel.isSwipeToNext = spy();
+
+        carousel.onTouchEnd(event as any);
+
+        expect(Carousel.isSwipeToNext).to.not.be.called;
+        expect(event.target.removeEventListener).to.be.calledWithExactly('touchstart', carousel.onTouchStart);
+        expect(event.target.removeEventListener).to.be.calledWithExactly('touchend', carousel.onTouchEnd);
+      });
+    });
+
+    describe('slideHandler()', () => {
+      it('should simply set slide number if speed is 0', () => {
+        const reset = carousel.resetCurrentSlideNum = spy();
+        const update = carousel.set = spy();
+
+        carousel.slideHandler(-1);
+
+        expect(reset).to.be.calledWithExactly(0, -1, 10);
+        expect(update).to.not.be.called;
+      });
+
+      it('should go to the specific slide when threshold is not hit and speed is not 0', () => {
+        carousel.state.settings.speed = 200;
+        const slide = 1;
+        const track = <any>{ addEventListener: spy(), removeEventListener: spy() };
+        carousel.refs = <any>{ track };
+        carousel.set = spy();
+        carousel.disableTransition = spy();
+
+        carousel.slideHandler(slide);
+
+        expect(carousel.set).to.be.calledWithExactly({ currentSlide: slide, transitioning: true });
+        expect(carousel.animationEndCallback).to.not.exist;
+        track.addEventListener.args[0][1]();
+        // tslint:disable-next-line:max-line-length
+        expect(track.addEventListener).to.be.calledWithExactly('transitionend', carousel.disableTransition, false);
+      });
+
+      describe('should reset to non-cloned slide when cloned slides are visible if speed is not 0', () => {
+        beforeEach(() => {
+          carousel.state.settings.speed = 100;
+          carousel.set = spy();
+        });
+
+        it('should be on edge', () => {
+          carousel.state.settings.slidesToShow = 3;
+          carousel.state.settings.slidesToScroll = 1;
+          carousel.state.currentSlide = 9;
+          const slide = 10;
+
+          carousel.slideHandler(slide);
+
+          expect(carousel.animationEndCallback).to.exist;
```

Comment:
Maybe `.to.be.a('function')`? This would pass for anything not null/undefined which I think is fairly broad since you want to check whether or not it's a callback.